src/components/forms/formComponents.js

```diff
@@ -74,14 +74,29 @@ export const formInputGeneric = (config) => {
   return div([
     input({
       id,
-      type: type || 'text',
+      type: type?.inputType || 'text',
       className: `form-control ${error ? 'errored' : ''}`,
       placeholder: placeholder || title,
-      value: formValue,
+      value: formValue.toString(),
       readOnly: readOnly,
       style: { ...styles.inputStyle, ...inputStyle },
       disabled: disabled,
-      onChange: (event) => onFormInputChange(config, event.target.value),
+      onChange: (event) => {
+        const value = event.target.value;
+        if (type?.inputType === 'number') {
```

Comment:
This parsing looks pretty complicated, especially since `onFormInputChange` calls `validateFormInput`. Perhaps the functionality could be moved into a `parseFormInput` function which is called after `validateFormInput`?